### PR TITLE
replace HahSet with fst::Set in classifier

### PR DESCRIPTION
**Pull request by MarinPostma**

### CODE REVIEW

**Review by LegendreM — APPROVED** [review]

File: src/analyzer.rs [hunk 1]
```diff
@@ -118,35 +122,35 @@ impl Default for AnalyzerConfig {
...
-pub struct Analyzer {
-    config: AnalyzerConfig,
+pub struct Analyzer<'a, A> {
+    config: AnalyzerConfig<'a, A>,
 }
 
-pub struct AnalyzedText<'a> {
+pub struct AnalyzedText<'a, A>
+where
+    A: AsRef<[u8]>
```
Comment (Kerollmops) [inline src/analyzer.rs:140]: Not sure but I think you don't need this bound here, only on implementations.

File: src/analyzer.rs [hunk 2]
```diff
@@ -235,7 +244,8 @@ mod test {
 
     #[test]
     fn test_simple_latin() {
-        let analyzer = Analyzer::new(AnalyzerConfig::default());
+        let stop_words = Set::from_iter([""].iter()).unwrap();
```
Comment (Kerollmops) [inline src/analyzer.rs:247]: There is a default impl, it's usually better than doing this by hand, here you create one empty key, do you really want that?

Comment (Kerollmops) [reply src/analyzer.rs:247]: It you wanted an empty iterator a `None as Option<&str>` would have been better.

File: src/token_classifier.rs [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 
 use crate::{Token, TokenKind};
 use crate::token::SeparatorKind;
 
 #[derive(Debug,  Clone)]
-pub struct TokenClassifier<'a> {
-    stop_words: &'a HashSet<String>,
+pub struct TokenClassifier<'a, A>
+where
+    A: AsRef<[u8]>
```
Comment (Kerollmops) [inline src/token_classifier.rs:10]: Same here and everywhere else, no need for this bound. :)

Comment (MarinPostma) [reply src/token_classifier.rs:10]: I will look at that! Thanks mops 😊

File: src/analyzer.rs (hunk 2 above)
Comment (MarinPostma) [reply src/analyzer.rs:247]: thanks, I was not sure how to deal with that :)

**Pull request merged**